OptimoveSDK/Sources/Classes/Optimobile/AnalyticsHelper.swift

```diff
@@ -226,17 +232,30 @@ final class AnalyticsHelper {
 
         let path = "/v1/app-installs/\(OptimobileHelper.installId)/events"
 
+        var err : Error? = nil
+        let networkBarrier = DispatchSemaphore(value: 0)
+
         eventsHttpClient.sendRequest(.POST, toPath: path, data: data, onSuccess: { _, _ in
-            if let err = self.pruneEventsBatch(context, eventIds) {
-                print("Failed to prune events batch: " + err.localizedDescription)
-                onSyncComplete?(err)
-                return
-            }
-            self.syncEvents(context: context, onSyncComplete)
+            networkBarrier.signal()
         }) { _, error, _ in
-            print("Failed to send events")
-            onSyncComplete?(error)
+            err = error
+            networkBarrier.signal()
```

Comment:
I noticed we're assigning the error to `err` in the failure callback but not logging it. Should we add a print statement or some form of logging here to help with debugging?

Reply:
Trying to avoid any new prints. (Used to have a logging mechanism that allowed the log level control)

Reply:
print() is only visible on the debugger I think, so it is only useful for developer feedback, so I'd vote to keep prints for SDK development purposes / app developer debug builds. WDYT?